alb-listeners.tf

```diff
@@ -1,39 +1,77 @@
-# If a certificate is provided, will redirect 80 to 443 and 443 will forward to target group
-resource "aws_lb_listener" "redirect_to_https" {
-  count             = var.certificate_arn != "" ? 1 : 0
+locals {
+  listener = {
+    http = { port = 80, protocol = "HTTP", action = { type = "forward" }}
+    http_redirect = { port = 80, protocol = "HTTP", action = { type = "redirect", port = 443, protocol = "HTTPS" }}
+    https = { port = 443, protocol = "HTTPS", action = { type = "forward" }}
+  }
+  listeners = length(var.listeners) == 0 && var.certificate_arn == "" ? [local.listener.http] : length(var.listeners) == 0 && var.certificate_arn != "" ? [local.listener.http_redirect, local.listener.https] : var.listeners
```

Comment:
I'd prob comment this line to say how it works.  "If the user didn't provide listeners, by default we do port 80, unless they provide a cert, then we do 443 and redirect 80=>443